pyspark_ai/tool.py

```diff
@@ -104,3 +104,73 @@ def _run(
 
     async def _arun(self, *args: Any, **kwargs: Any) -> Any:
         raise NotImplementedError("ListTablesSqlDbTool does not support async")
+
+
+class VectorSearchUtil:
+    @staticmethod
+    def vector_similarity_search(
+        col_lst: list,
+        vector_store_path: Optional[str],
+        search_text: str,
+        col: str,
+        temp_name: str,
+    ):
+        from langchain.vectorstores import FAISS
+        from langchain.embeddings import HuggingFaceBgeEmbeddings
+        import os
+
+        if vector_store_path and os.path.exists(vector_store_path):
+            vector_db = FAISS.load_local(vector_store_path, HuggingFaceBgeEmbeddings())
+        else:
+            vector_db = FAISS.from_texts(col_lst, HuggingFaceBgeEmbeddings())
+
+            if vector_store_path:
+                vector_db.save_local(vector_store_path)
+
+        docs = vector_db.similarity_search(search_text)
+        return docs[0].page_content
+
+
+class SimilarValueTool(BaseTool):
+    """Tool for finding the semantically closest word to a keyword from a vector database."""
```

Comment:
I am not sure if we should make it more specific here by saying: Tool for finding the column value which is closet to the input text.